docs/topics/development/troubleshooting_and_debugging.md

```diff
@@ -0,0 +1,176 @@
+
+     ```sh
+     make down
+     make up
+     ```
+
+3. **Missing Dependencies**:
+   - **Issue**: Missing Python or Node.js dependencies.
+   - **Solution**: Ensure that all dependencies are installed by running the following command:
+
+     ```sh
+     make update_deps
+     ```
+
+4. **Locale Compilation Issues**:
+   - **Issue**: Locales are not compiling correctly.
+   - **Solution**: Run the locale compilation command and check for any errors:
+
+     ```sh
+     make compile_locales
+     ```
+
+5. **Permission Issues**:
+   - **Issue**: Permission errors when accessing files or directories.
+   - **Solution**: Ensure that the `olympia` user has the correct permissions. Use `chown` or `chmod` to adjust permissions if necessary.
+
+## Debugging
+
+The Docker setup uses `supervisord` to run the Django runserver. This allows you to access the management server from a shell to run things like `ipdb`.
+
+### Using `ipdb`
+
+To debug with `ipdb`, add a line in your code at the relevant point:
+
+```python
+import ipdb; ipdb.set_trace()
+```
+
+Next, connect to the running web container:
+
+```sh
+make debug
+```
+
+This command brings the Django management server to the foreground, allowing you to interact with `ipdb` as you normally would. To quit, type `Ctrl+c`.
+
+Example session:
+
+```sh
+$ make debug
+docker exec -t -i olympia_web_1 supervisorctl fg olympia
+:/opt/rh/python27/root/usr/lib/python2.7/site-packages/celery/utils/__init__.py:93
+11:02:08 py.warnings:WARNING /opt/rh/python27/root/usr/lib/python2.7/site-packages/jwt/api_jws.py:118: DeprecationWarning: The verify parameter is deprecated. Please use options instead.
+'Please use options instead.', DeprecationWarning)
+:/opt/rh/python27/root/usr/lib/python2.7/site-packages/jwt/api_jws.py:118
+[21/Oct/2015 11:02:08] "PUT /en-US/firefox/api/v4/addons/%40unlisted/versions/0.0.5/ HTTP/1.1" 400 36
+Validating models...
+
+0 errors found
+October 21, 2015 - 13:52:07
+Django version 1.6.11, using settings 'settings'
+Starting development server at http://0.0.0.0:8000/
+Quit the server with CONTROL-C.
+[21/Oct/2015 13:57:56] "GET /static/img/app-icons/16/sprite.png HTTP/1.1" 200 3810
+13:58:01 py.warnings:WARNING /opt/rh/python27/root/usr/lib/python2.7/site-packages/celery/task/sets.py:23: CDeprecationWarning:
+    celery.task.sets and TaskSet is deprecated and scheduled for removal in
+    version 4.0. Please use "group" instead (see the Canvas section in the userguide)
+
+"""
+:/opt/rh/python27/root/usr/lib/python2.7/site-packages/celery/utils/__init__.py:93
+> /code/src/olympia/browse/views.py(148)themes()
+    147     import ipdb;ipdb.set_trace()
+--> 148     TYPE = amo.ADDON_THEME
+    149     if category is not None:
+
+ipdb> n
+> /code/src/olympia/browse/views.py(149)themes()
+    148     TYPE = amo.ADDON_THEME
+--> 149     if category is not None:
+    150         q = Category.objects.filter(application=request.APP.id, type=TYPE)
+
+ipdb>
+```
```

Comment:
This entire section is (unfortunately) obsolete. `make debug` is currently broken, see https://github.com/mozilla/addons/issues/1808

Adding a breakpoint is still useful when you're running tests or using the shell/a management command though.

Reply:
The whole doc or which section in particular?

Reply:
Better?

Reply:
Yes (eventually it'd be good to make `make debug` work again, but in the meantime let's ensure the docs are accurate)